File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let hash = {}
    for (let num of nums){
        if (hash[num] !== undefined){
            return true;
        } else{
            hash[num] = false;
        }
    }
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
from collections import defaultdict
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        count = defaultdict(int)
        for num in nums: 
            if count[num] == 1: 
                return True
            count[num] += 1
            
        return False
        # dupSet = set()
        # for num in nums: 
        #     if num in dupSet: 
        #         return True
        #     dupSet.add(num)
        # return False